alica_engine/include/engine/AlicaContext.h

```diff
@@ -248,15 +286,45 @@ class AlicaContext
     /**
      * Returns agent id for this alica context.
      *
-     * @return Object representing id.
+     * @return Object representing id of local agent.
      */
-    AgentID getLocalAgentId() const;
+    essentials::IdentifierConstPtr getLocalAgentId() const;
 
     /**
      * Execute one step of engine synchronously
      */
     void stepEngine();
 
+    /**
+     * If present, returns the ID corresponding to the given prototype.
+     * Otherwise, it creates a new one, stores, and returns it.
+     *
+     * This method can be used, e.g., for passing an int and receiving
+     * a pointer to a corresponding ID object.
+     * @tparam Prototype Datatype of given prototyp id
+     * @param idPrototype The prototyp id
+     * @param type Type of requested ID Object
+     * @return ID object
+     */
+    template <class Prototype>
+    essentials::IdentifierConstPtr getID(Prototype& idPrototype, uint8_t type = essentials::Identifier::UUID_TYPE);
```

Comment:
@StephanOpfer did you just merge without addressing the comments?

Reply:
As it was approved, it merged it and planned to address the comments in an extra PR. I am not sure, whether we can live without this API - need to check if there is a solution.
One problem I have in mind: 
IDs are sent and received in message. In order to deserialise these IDs, the module that implements the ALICA Communication interface needs access to the IDManager because there only should be one instance per ID in memory.